py3status/modules/deadbeef.py

```diff
@@ -3,95 +3,101 @@
-                                   '--nowplaying',
-                                   self.delimiter.join(['%a',
-                                                        '%t',
-                                                        '%l',
-                                                        '%e',
-                                                        '%y',
-                                                        '%n'])])
-
-            if status == 'nothing':
-                return self._empty_response()
-
-            # split properties using special delimiter
-            parts = status.split(self.delimiter)
-            if len(parts) == 6:
-                artist, title, length, elapsed, year, tracknum = parts
-            else:
-                return self._error_response(self.py3.COLOR_BAD)
-
-            response = {
+    def deadbeef(self):
+        color = self.py3.COLOR_BAD
+        artist = isplaying = length = playback_time = title = \
+            tracknumber = year = ""
+
+        if not self._is_running():
+            return {
                 'cached_until': self.py3.time_in(self.cache_timeout),
-                'full_text': self.py3.safe_format(self.format,
-                                                  dict(artist=artist,
-                                                       title=title,
-                                                       length=length,
-                                                       elapsed=elapsed,
-                                                       year=year,
-                                                       tracknum=tracknum)
-                                                  )
+                'color': color,
+                'full_text': '',
             }
-            return response
-        except:
-            return self._error_response(self.py3.COLOR_BAD)
+
+        # mix metadata fields in with the delimiters
+        fmt = self.delimiter.join([
+            '%artist%', '%isplaying%', '%length%', '%playback_time%',
+            '%title%', '%tracknumber%', '%year%'
+        ])
+        # run command and get output
+        out = subprocess.check_output(['deadbeef', '--nowplaying-tf', fmt])
+        out = out.decode('utf-8')
+
+        # get metadata results out of the string using the delimiters
+        if out != 'nothing':
+            artist, isplaying, length, playback_time, title, tracknumber, \
+                year = out.split(self.delimiter)
+
+            if title:
+                if isplaying == "1":
+                    color = self.py3.COLOR_GOOD
+                else:
+                    color = self.py3.COLOR_DEGRADED
+
+        # response
+        deadbeef = self.py3.safe_format(self.format, dict(
+            artist=artist, isplaying=isplaying, length=length,
+            playback_time=playback_time, title=title,
+            tracknumber=tracknumber, year=year))
+
+        deadbeef = deadbeef.strip().strip('-').strip(':')
```

Comment:
We remove this when we have formatter. It omits ` - ` if we are not playing a song. I added ` : ` in case somebody wanted to use `aritst: title` or something like that.